File: src/RechercheService/contracts/enchereCreated.js
const mongoose = require("mongoose");

const enchereCreatedSchema = new mongoose.Schema({
  id: { type: String, required: true },
  reservePrice: { type: Number, default: 0 },
  seller: String,
  winner: String,
  soldAmount: Number,
  currentHighBid: Number,
  createdAt: Date,
  updatedAt: Date,
  auctionEnd: Date,
  auctionStart: Date,
  status: String,
  make: String,
  name: String,
  year: Number,
  color: String,
  category: String,
  state: String,
  description: String,
  imageUrl: String,
});

const EnchereCreated = mongoose.model("EnchereCreated", enchereCreatedSchema);

module.exports = EnchereCreated;

File: src/Contracts/EnchereCreated.cs
namespace Contracts;

public class EnchereCreated
{
    public Guid Id { get; set; }
    public int ReservePrice { get; set; }
    public string Seller { get; set; }
    public string Winner { get; set; }
    public int SoldAmount { get; set; }
    public int CurrentHighBid { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public DateTime AuctionStart { get; set; }
    public DateTime AuctionEnd { get; set; }
    public string Statut { get; set; }
    public string Make { get; set; }
    public string Name { get; set; }
    public int Year { get; set; }
    public string Color { get; set; }
    public string Category { get; set; }
    public string State { get; set; }
    public string Description { get; set; }
    public string ImageUrl { get; set; }
}
